WIP - CFEP to spend conda-forge funds (#1)
trallard/cfep: cfep-26.md

<table>
<tr><td> Title </td><td> Spending conda-forge project funds </td>
<tr><td> Status </td><td> Draft  </td></tr>
<tr><td> Author(s) </td><td> Tania Allard &lt;taniar.allard@gmail.com&gt;, Jaime Rodríguez Guerra &lt;xxxxxxxxx@gmail.com&gt;</td></tr>
<tr><td> Created </td><td> Jun 13, 2025</td></tr>
<tr><td> Updated </td><td> Jun 13, 2025</td></tr>
<tr><td> Discussion </td><td> NA </td></tr>

> <tr><td> Implementation </td><td> link to the PR for the implementation, NA if not availble </td></tr>

Reviewer: ```suggestion
<tr><td> Implementation </td><td> NA </td></tr>
```

</table>

## Abstract

This CFEP proposes a policy for how to spend unrestricted conda-forge project funds.

> It aims to provide a set of transparent principles and guidelines for decision-making

Reviewer: ```suggestion
It aims to provide a set of principles and guidelines for decision-making
```

(reduce redundancy below)

around the transparent use of and administration of these funds, ensuring that they are used
in a way that benefits the conda-forge community.

## Motivation

conda-forge is a NumFOCUS (Austin, TX based non-profit 501(c)(3)) fiscally sponsored project,
as such, the conda-forge project has to align with accounting and legal regulations for US-based non-profits.
Under these considerations, non-profit donations are classified into two categories:

> **unrestricted funds** which may be used for any legal purpose appropriate to the
> organization and **restricted funds**, monies set aside for a particular purpose or
> program (for example, project-specific deliverables, educational programs, etc.).

Reviewer: ```suggestion

- **unrestricted funds** which may be used for any legal purpose appropriate to the
organization.
- **restricted funds**, money set aside for a particular purpose or
program (for example, project-specific deliverables, educational programs, etc.).
```

as a list for clarity

> Up to now, conda-forge remains a largely volunteer-run project without significant

Reviewer: Just to have some numbers, what amount would be significant?

Author: Hand-wavy: something around enough funds to hire a developer in residence. If you need more precise numbers, I can dig something up.

amounts of unrestricted funds (either from individual donations or corporate sponsorships).
Thus, robust policies for spending these funds have not yet been critical to the project.
However, as the conda-forge project grows there is also the potential for increased

> Donations and grants are going directly to the project.

Reviewer: ```suggestion
donations and grants going directly to the project.
```

- Guidelines to manage conda-forge project funds.

Out-of-scope items:

- Decision-making around spending project funds on a specific project or activity.
- Principles and processes for spending restricted funds earmarked for
  specific purposes or projects.
  This includes most of the grant funding, which is usually earmarked for certain
  activities/deliverables and goes to an Institutional Partner rather than directly to
  the conda-forge project and companies or institutions funding specific features.
  *For grants, though, we expect the research/project leader and funded team to align their
  work with the needs of conda-forge and be receptive to feedback from conda-forge
  core team members, maintainers, and contributors.*

## Principles for spending project funds

conda-forge is and will likely remain a project with many more volunteer contributors
than funded people.
As such, it is critical to ensure that those funded people operate in ways that attract
more volunteers and enhance their participation in the project.

From a top-level the approach for spending conda-forge project funds is to:

1. Define *what* to fund

> 2. Identify a suitable person to carry out the work

Reviewer: ```suggestion
2. Identify a suitable person or team to carry out the work
```

3. Determine a fair compensation for the work

### Defining what to fund

When assessing what needs funding, the following guidelines should be considered:

1. The work should be aligned with the conda-forge project goals and priorities.

> 2. Fund things that are important *and* that, otherwise, won't get done.

Reviewer: ```suggestion
2. Fund things that are important *and* that, otherwise, wouldn't get done.
```

3. Plan for sustainability, that is, ensuring that the work can be maintained over time
   without requiring constant funding.
4. Prioritize work that will benefit the conda-forge community as a whole (contributors,
   maintainers, feedstock maintainers, users, etc.).
5. All types of work are considered critical to a project's sustainability,
   including but not limited to:
   - Development work (e.g., improving the conda-forge infrastructure, tools, etc.)
   - Community building (e.g., organizing events, workshops, etc.)
   - Documentation and educational materials (e.g., tutorials, guides, etc.)
   - Outreach and communication (e.g., engaging with users, etc.)

>    - Contributor experience and governance (e.g., improving the onboarding process,

Reviewer: ```suggestion
   - Contributor experience and governance (e.g., improving onboarding and offboarding,
```

     supporting contributors, streamlining processes, etc.)
6. Consider the impact of the work on other maintainers or core members; consider paid
   time for others to review work if this is expected to take a significant amount of
   time and effort vs. solely increasing the load on other volunteers.

When considering what needs funding, principle 2 implies a focus on the boring/painful,
yet essential work that nobody likes doing and necessary structural changes to the project
that are too large to be done by a volunteer in a reasonable amount of time.

Funding can be used for items beyond people's time, for example, expenses for in-person

> developer sprints, hosted hardware for benchmarking or specialized development work.

Reviewer: ```suggestion
developer sprints, hosted hardware for packaging or benchmarking, or specialized development work.
```

> ### Whom to pay

Reviewer: Anything worth adding about "the ability to receive the funds in a legal way"?

Author: I still need to add some bits, but yes, I, I will add a note about ensuring these are also aligned with the requirements inherited from our fiscal host.

<table>
<tr><td> Title </td><td> Spending conda-forge project funds </td>
<tr><td> Status </td><td> Draft  </td></tr>

> <tr><td> Author(s) </td><td> Tania Allard &lt;taniar.allard@gmail.com&gt;, Jaime Rodríguez Guerra &lt;xxxxxxxxx@gmail.com&gt;</td></tr>

Author: @jaimergp to confirm email

Reviewer: ```suggestion
<tr><td> Author(s) </td><td> Tania Allard &lt;taniar.allard@gmail.com&gt;, Jaime Rodríguez Guerra &lt;jrodriguez@quansight.com&gt;</td></tr>
```

unrestricted funds can be used to support the community and its goals. Additionally,
having such principles and processes in place will make it easier to attract and manage
donations and other types of contributions moving forward.

## Scope

The following items are in scope for this CFEP:

- Principles of spending project funds (what to fund, and who to pay for this).
- Processes for decision-making around spending project funds and fund management.
- Guidelines to manage conda-forge project funds.

Out of scope items:

- Decision-making around spending project funds on a specific project or activity.
- Principles and processes for spending restricted funds, which are earmarked for
  specific purposes or projects.
  This includes most of the grant funding, which is usually earmarked for certain
  activities/deliverables and goes to an Institutional Partner rather than directly to
  the conda-forge project, and companies or institutions funding specific features.
  *For grants though, we expect the research/project leader and funded team to align their
  work with the needs of conda-forge and be receptive to feedback from conda-forge

>   core team members, maintainers, and contributors.*

Author: I'm unsure if there is better wording or if these expectations are already set elsewhere, @jaimergp.
